### PR TITLE
🐛 Flaky ClusterResourceSet test fix

### PR DESCRIPTION
**Pull request by sedefsavas**
**What this PR does / why we need it**:

The frequent flakiness we observe in CRS test is due to the timeout value set in this test.
Locally this is unreproducible as 10 seconds is enough for ClusterResourceSetBinding creation, but maybe test infra became slower and we are hitting this limit. Increased timeout value, we should observe this less frequent.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

### CODE REVIEW

**Comment by sedefsavas** [comment]
The issue was locally unreproducible so testing the fix here.

File: exp/addons/controllers/clusterresourceset_controller_test.go [hunk 1]
```diff
@@ -289,14 +285,18 @@ metadata:
...
 				Name:      newCMName,
 				Namespace: defaultNamespaceName,
 			},
 			Data: map[string]string{},
 		}
-		Expect(testEnv.Create(ctx, testConfigmap)).To(Succeed())
+		Expect(testEnv.Create(ctx, newConfigmap)).To(Succeed())
+		defer func() {
+			Expect(testEnv.Delete(ctx, newConfigmap)).To(Succeed())
+		}()
```
Comment (JoelSpeed) [inline exp/addons/controllers/clusterresourceset_controller_test.go:298]: Does this get run even if a later assertion fails?

File: exp/addons/controllers/clusterresourceset_controller_test.go [hunk 2]
```diff
@@ -242,10 +239,9 @@ metadata:
 		}, timeout).Should(BeTrue())
 	})
 	It("Should reconcile a ClusterResourceSet when a resource is created that is part of ClusterResourceSet resources", func() {
-		labels := map[string]string{"foo2": "bar2"}
-		newCMName := "test-configmap3"
+		newCMName := fmt.Sprintf("test-configmap-%s", util.RandomString(6))
```
Comment (JoelSpeed) [inline exp/addons/controllers/clusterresourceset_controller_test.go:242]: You could use the metadata `GenerateName` to do this for you on create, though I guess that means you have to create the configmap before referencing it elsewhere which may make the tests more complex

File: exp/addons/controllers/clusterresourceset_controller_test.go [hunk 3]
```diff
@@ -318,11 +318,18 @@ metadata:
 			}
 			return errors.Errorf("ClusterResourceSet binding does not have any resources matching %q: %v", newCMName, binding.Spec.Bindings)
 		}, timeout).Should(Succeed())
-		Expect(testEnv.Delete(ctx, testConfigmap)).To(Succeed())
+
+		By("Verifying ClusterResourceSetBinding is deleted when its cluster owner reference is deleted")
```
Comment (JoelSpeed) [inline exp/addons/controllers/clusterresourceset_controller_test.go:322]: Is the garbage collector running on this test suite? I didn't think it was normally being run in envtest style environments. Or is the CRS controller doing this cleanup?

Comment (sedefsavas) [reply exp/addons/controllers/clusterresourceset_controller_test.go:322]: `ClusterResourceSetBinding` is deleted by CRS controller.

File: exp/addons/controllers/clusterresourceset_controller_test.go (hunk 1 above)
Comment (sedefsavas) [reply exp/addons/controllers/clusterresourceset_controller_test.go:298]: Yes, it gets run.

File: exp/addons/controllers/clusterresourceset_controller_test.go (hunk 2 above)
Comment (sedefsavas) [reply exp/addons/controllers/clusterresourceset_controller_test.go:242]: Right, I prefer to keep it as is.

**Comment by sedefsavas** [comment]
This is ready to go.

**Pull request merged**